source_gen/lib/src/builder.dart

```diff
@@ -94,8 +102,7 @@ class _Builder extends Builder {
     // library/part definitions because users expect some files to be skipped
     // therefore they do not have "library".
     if (generatedOutputs.isEmpty) return;
-    final outputId = buildStep.inputId.changeExtension(_generatedExtension);
-
+    final outputId = buildStep.allowedOutputs.single;
```

Comment:
There might be more than one if there are `additionalOutputExtensions`. Presumably the ordering is kept here though so you could just grab the first?

```suggestion
    final outputId = buildStep.allowedOutputs.first;
```

If this didn't break tests we must not be testing `additionalOutputExtensions`?